pandas/tests/test_window.py

```diff
@@ -1719,66 +1475,16 @@ def test_ewm_domain_checks(self):
         s.ewm(alpha=1.0)
         pytest.raises(ValueError, s.ewm, alpha=1.1)
 
-    def test_ew_empty_arrays(self):
-        arr = np.array([], dtype=np.float64)
+    def test_ew_empty_series(self):
+        vals = pd.Series([], dtype=np.float64)
 
-        funcs = [mom.ewma, mom.ewmvol, mom.ewmvar]
+        ewm = vals.ewm(3)
+        funcs = ['mean', 'vol', 'var']
         for f in funcs:
-            with catch_warnings(record=True):
-                result = f(arr, 3)
-            tm.assert_almost_equal(result, arr)
-
-    def _check_ew(self, func, name=None):
-        with catch_warnings(record=True):
-            self._check_ew_ndarray(func, name=name)
-        self._check_ew_structures(func, name=name)
-
-    def _check_ew_ndarray(self, func, preserve_nan=False, name=None):
-        result = func(self.arr, com=10)
-        if preserve_nan:
-            assert (np.isnan(result[self._nan_locs]).all())
-
-        # excluding NaNs correctly
-        arr = randn(50)
-        arr[:10] = np.NaN
-        arr[-10:] = np.NaN
-        s = Series(arr)
-
-        # check min_periods
-        # GH 7898
-        result = func(s, 50, min_periods=2)
-        assert np.isnan(result.values[:11]).all()
-        assert not np.isnan(result.values[11:]).any()
-
-        for min_periods in (0, 1):
-            result = func(s, 50, min_periods=min_periods)
-            if func == mom.ewma:
-                assert np.isnan(result.values[:10]).all()
-                assert not np.isnan(result.values[10:]).any()
-            else:
-                # ewmstd, ewmvol, ewmvar (with bias=False) require at least two
-                # values
-                assert np.isnan(result.values[:11]).all()
-                assert not np.isnan(result.values[11:]).any()
+            result = getattr(ewm, f)()
+            tm.assert_almost_equal(result, vals)
 
-            # check series of length 0
-            result = func(Series([]), 50, min_periods=min_periods)
-            tm.assert_series_equal(result, Series([]))
-
-            # check series of length 1
-            result = func(Series([1.]), 50, min_periods=min_periods)
-            if func == mom.ewma:
-                tm.assert_series_equal(result, Series([1.]))
-            else:
-                # ewmstd, ewmvol, ewmvar with bias=False require at least two
-                # values
-                tm.assert_series_equal(result, Series([np.NaN]))
-
-        # pass in ints
-        result2 = func(np.arange(50), span=10)
-        assert result2.dtype == np.float_
-
-    def _check_ew_structures(self, func, name):
+    def _check_ew(self, name=None):
```

Comment:
Previously this function consisted of `_check_ew_ndarray` and `_check_ew_structures` subfunctions to test both interfaces. You removed `_check_ew_ndarray` and only kept the content of `_check_ew_structures` as this is testing the new interface. 
But, as far as I can see, the *actual* testing was done in `_check_ew_ndarray`  (`_check_ew_structures` is only testing the output class type). Shouldn't this testing be kept? (changed to the new interface)

(might good be I am missing something, or that it is covered in other tests)

Reply:
I think you are right: So rather than test this function call, test the ``ewm`` method call on a series. I'll look into it tomorrow.